app/lib/import_export.rb

```diff
@@ -692,222 +368,62 @@ def extract_assessments_details(patients_identifiers, assessments, fields)
     exported_data[data_type]&.each do |record|
-      sheet.add_row(fields.map { |field| record[field] }, { types: Array.new(fields.length, :string) })
-    end
-  end
-
-  # Gets data for this batch of patients that may not have already been present in the export config (such as specific symptoms).
-  def get_field_data(patients_group, config)
-    data = config[:data].deep_dup
-
-    # Update the checked data (used for obtaining values) with race information
-    update_checked_race_data(data)
-
-    # Update the header data (used for obtaining column names)
-    update_headers(data)
-
-    # Update the checked and header data for assessment symptoms
-    # NOTE: this must be done after updating the general headers above
-    update_assessment_symptom_data(patients_group, data)
-
-    data
-  end
-
-  # Finds the symptoms needed for the reports columns
-  def update_assessment_symptom_data(patients_group, data)
-    # Don't update if assessment symptom data isn't needed
-    return unless data.dig(:assessments, :checked).present? && data[:assessments][:checked].include?(:symptoms)
-
-    symptom_names_and_labels = patients_group.joins(assessments: [{ reported_condition: :symptoms }]).pluck('symptoms.name, symptoms.label').uniq.sort
-
-    data[:assessments][:checked].delete(:symptoms)
-    data[:assessments][:checked].concat(symptom_names_and_labels.map(&:first).map(&:to_sym))
-
-    data[:assessments][:headers].delete('Symptoms Reported')
-    data[:assessments][:headers].concat(symptom_names_and_labels.map(&:second))
-  end
-
-  # Finds the race values that should be included if the Race (All Race Fields) option is checked in custom export
-  def update_checked_race_data(data)
-    # Don't update if patient data isn't needed or race data isn't checked
-    return unless data.dig(:patients, :checked).present? && data[:patients][:checked].include?(:race)
-
-    # Replace race field with actual race fields
-    race_index = data[:patients][:checked].index(:race)
-    data[:patients][:checked].delete(:race)
-    data[:patients][:checked].insert(race_index, *PATIENT_RACE_FIELDS)
-  end
-
-  # Update the header data (used for obtaining column names)
-  def update_headers(data)
-    # Populate the headers if they're not already set (which is the case with the custom exports)
-    CUSTOM_EXPORT_OPTIONS.each_key do |data_type|
-      next unless data.dig(data_type, :checked).present? && data.dig(data_type, :headers).blank?
-
-      data[data_type][:headers] = data[data_type][:checked].map { |field| ALL_FIELDS_NAMES.dig(data_type, field) }
+      # fast_excel unfortunately does not provide a method to modify the @last_row_number class variable so it needs to be manually kept track of
+      last_row_num += 1
+      fields.each_with_index do |field, col_index|
+        # write_string is used instead of append_row because it does not provide the capability to write data as strings
```

Comment:
the documentation is not very clear about this and I had to do some digging into the source code to figure out what was going on under the hood:

(append_row)[https://github.com/Paxa/fast_excel/blob/8fafe6f20cf51c683314a589f1597113d8704b0a/lib/fast_excel.rb#L501] calls (write_row)[https://github.com/Paxa/fast_excel/blob/8fafe6f20cf51c683314a589f1597113d8704b0a/lib/fast_excel.rb#L422] which calls (write_value)[https://github.com/Paxa/fast_excel/blob/8fafe6f20cf51c683314a589f1597113d8704b0a/lib/fast_excel.rb#L445] which tries to be smart by detecting the data type of each value and writing it to excel as that data type format

I tried casting all our data to strings but was unsuccessful but even if I was, calling `write_string` directly is more performant as it would skip all the if/elsif/else logic from (write_value)[https://github.com/Paxa/fast_excel/blob/8fafe6f20cf51c683314a589f1597113d8704b0a/lib/fast_excel.rb#L445]